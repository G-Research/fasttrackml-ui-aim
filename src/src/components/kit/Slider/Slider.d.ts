import React from 'react';

import { SliderProps } from '@material-ui/core';

export interface ISliderProps extends SliderProps {
  containerClassName?: string;
  style?: React.CSSProperties;
  onChange?: (event: React.ChangeEvent<{}>, value: number | number[]) => void;
  prevIconNode?: React.ReactNode;
  nextIconNode?: React.ReactNode;
  label?: React.ReactNode;
}
