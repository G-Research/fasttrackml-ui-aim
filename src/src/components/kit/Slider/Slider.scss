@use 'src/styles/abstracts' as *;

.Slider {
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  .MuiSlider-root {
    padding: 0;
    height: 0.25rem;
    .MuiSlider-rail {
      height: 0.25rem;
      border-radius: $border-radius-sm;
      opacity: 1;
      background: $cuddle-50;
    }
    .MuiSlider-track {
      height: 0.25rem;
      border-radius: $border-radius-sm;
      background: $primary-color;
    }
    .MuiSlider-thumb {
      background: $white;
      border: 0.09375rem solid $primary-color;
      margin-top: -0.25rem;
      margin-left: -$space-xxs;
      .MuiSlider-valueLabel {
        left: calc(-50% - 0.625rem);
        top: -1.0625rem;
      }
      .MuiSlider-valueLabel > span {
        width: 2.3125rem;
        height: 1.1875rem;
        border: 0.0625rem solid $primary-color;
        background: $white;
        border-radius: 0.4375rem;
        transform: unset;
        span {
          font-size: 0.625rem;
          font-weight: $font-600;
          color: $primary-color;
          transform: unset;
          text-overflow: ellipsis;
          max-width: 100%;
          overflow: hidden;
          padding: $space-xxxxs $space-xxxs;
        }
      }
      &.Mui-focusVisible {
        box-shadow: 0 0 0 0.125rem rgba(20, 115, 230, 0.2);
      }
      &:hover {
        box-shadow: 0 0 0 0.125rem rgba(20, 115, 230, 0.2);
      }
      &.MuiSlider-active {
        box-shadow: 0 0 0 0.125rem rgba(20, 115, 230, 0.2);
      }
      &::after {
        top: 0px;
        bottom: 0px;
        left: 0px;
        right: 0px;
      }
    }
  }
  .Mui-disabled {
    .MuiSlider-track {
      background: #a7b7cd;
    }
    .MuiSlider-thumb {
      margin-top: toRem(-2px);
      margin-left: toRem(-4px);
    }
  }
  .Mui-disabled {
    .MuiSlider-track {
      background: #a7b7cd;
    }
    .MuiSlider-thumb {
      margin-top: toRem(-2px);
      margin-left: toRem(-4px);
    }
  }
}
