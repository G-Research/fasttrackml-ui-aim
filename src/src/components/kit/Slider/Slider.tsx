import React from 'react';

import { Slider as MaterialSlider } from '@material-ui/core';

import ErrorBoundary from 'components/ErrorBoundary/ErrorBoundary';

import { ISliderProps } from '.';

import './Slider.scss';

function Slider({
  containerClassName = '',
  style = {},
  prevIconNode = null,
  nextIconNode = null,
  label = null,
  ...rest
}: ISliderProps): React.FunctionComponentElement<React.ReactNode> {
  return (
    <ErrorBoundary>
      <div className={`Slider ${containerClassName}`} style={style}>
        {label}
        {prevIconNode && prevIconNode}
        <MaterialSlider {...rest} />
        {nextIconNode && nextIconNode}
      </div>
    </ErrorBoundary>
  );
}

Slider.displayName = 'Slider';

export default React.memo<ISliderProps>(Slider);
