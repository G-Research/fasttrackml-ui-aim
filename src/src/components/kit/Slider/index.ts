import Slider from './Slider';

export * from './Slider.d';

export default Slider;
