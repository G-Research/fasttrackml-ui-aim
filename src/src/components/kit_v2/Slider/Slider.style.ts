import * as SliderPrimitive from '@radix-ui/react-slider';

import { styled } from 'config/stitches';

const SliderRoot = styled(SliderPrimitive.Root, {
  position: 'relative',
  display: 'flex',
  ai: 'center',
  userSelect: 'none',
  touchAction: 'none',
  zIndex: '$1',
  '&[data-orientation="horizontal"]': {
    height: 10,
  },
  '&[data-orientation="vertical"]': {
    fd: 'column',
    width: 10,
    height: 100,
  },
});

const SliderMark = styled('span', {
  position: 'absolute',
  display: 'flex',
  ai: 'center',
  jc: 'center',
  bc: '#ACB2BC',
  width: 1,
  height: 6,
  zIndex: -1,
  br: '$1',
  cursor: 'pointer',
  '&[data-active=true]': {
    bc: '$primary100',
  },
  '&[data-disabled]': {
    pointerEvents: 'none',
    bc: '#CED1D7',
  },
  '&:before': {
    content: '""',
    position: 'absolute',
    top: -6,
    left: -1,
    width: 4,
    height: 20,
    bc: 'transparent',
    br: '$1',
  },
});

const SliderTrack = styled(SliderPrimitive.Track, {
  bc: '#CED1D7',
  position: 'relative',
  fg: 1,
  br: '$1',
  size: '$5',
  '&[data-orientation="horizontal"]': { height: 2 },
  '&[data-orientation="vertical"]': { width: 2 },
});

const SliderRange = styled(SliderPrimitive.Range, {
  position: 'absolute',
  bc: '$primary100',
  br: '$1',
  height: '100%',
});

const SliderThumb = styled(SliderPrimitive.Thumb, {
  position: 'relative',
  display: 'block',
  size: '10px',
  bc: '$primary100',
  br: '$5',
  '&:parent': {
    zIndex: '$10',
  },
  '&:hover, &:focus': {
    '&:after': {
      content: '""',
      position: 'absolute',
      top: 3,
      left: 3,
      width: 4,
      height: 4,
      bc: 'white',
      br: 2,
    },
  },
  '&:focus': {
    outline: 'none',
    bs: '0 0 0 2px $colors$primary20',
  },
});

const SliderLabel = styled('span', {
  position: 'absolute',
  width: '$1',
  height: '16px',
  top: '14px',
  left: '-5px',
  display: 'flex',
  ai: 'center',
  jc: 'center',
  bc: 'white',
  br: '$3',
  fontSize: '$1',
  border: '1px solid $colors$primary100',
});

export {
  SliderRoot,
  SliderMark,
  SliderTrack,
  SliderRange,
  SliderThumb,
  SliderLabel,
};
