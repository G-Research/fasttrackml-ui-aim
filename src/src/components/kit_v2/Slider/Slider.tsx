import React from 'react';
import _ from 'lodash-es';

import Text from '../Text';

import { ISliderProps } from './Slider.d';
import {
  SliderLabel,
  SliderMark,
  SliderRange,
  SliderRoot,
  SliderThumb,
  SliderTrack,
} from './Slider.style';

//TODO: Fix the thumb label overlapping issue
/**
 * Slider component
 * @param {number} min - min value
 * @param {number} max - max value
 * @param {number[]} value - value
 * @param {number[]} defaultValue - default value
 * @param {number[]} marks - marks
 * @param {function} onValueChange - onValueChange handler
 * @param {ISliderProps} rest - rest props
 * @returns {React.FunctionComponentElement<React.ReactNode>}
 * @example
 * <Slider min={0} max={100} value={[10, 20]} defaultValue={[10]} />
 */
const Slider = React.memo(
  ({
    min = 0,
    max = 100,
    value = [],
    defaultValue = [10],
    marks = [],
    onValueChange,
    ...props
  }: ISliderProps): React.FunctionComponentElement<React.ReactNode> => {
    const [sliderValue, setSliderValue] =
      React.useState<number[]>(defaultValue);

    React.useEffect(() => {
      if (value.length > 0 && !_.isEqual(value, sliderValue)) {
        setSliderValue(value);
      }
      // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [value]);

    const onChange = React.useCallback((value: number[]) => {
      setSliderValue(value);
      if (onValueChange) {
        onValueChange(value);
      }
      // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    function getMarkPosition(mark: number) {
      // The mark position is calculated based on the min and max values
      const markPosition = ((mark - min) / (max - min)) * 100;

      // The coefficient is needed to align the mark with the thumb
      const calcCoif: number = 5 - markPosition * 0.1;
      return `calc(${markPosition}% + ${calcCoif}px)`;
    }

    return (
      <SliderRoot
        value={sliderValue}
        defaultValue={defaultValue}
        min={min}
        max={max}
        onValueChange={onChange}
        data-testid='slider'
        {...props}
      >
        <SliderTrack className='SliderTrack'>
          <SliderRange className='SliderRange' />
        </SliderTrack>
        {sliderValue.map((value: number) => (
          <SliderThumb className='SliderThumb' key={value}>
            <SliderLabel className='SliderLabel'>{value}</SliderLabel>
          </SliderThumb>
        ))}

        {marks.length > 0 &&
          marks.map((mark) => {
            return (
              <SliderMark
                key={mark.value}
                data-active={
                  sliderValue?.length > 1
                    ? mark.value >= sliderValue?.[0] &&
                      mark.value <= sliderValue[sliderValue.length - 1]
                    : mark.value <= sliderValue?.[0]
                }
                css={{ left: `${getMarkPosition(mark.value)}` }}
              >
                <Text size='$1' css={{ marginTop: 33 }}>
                  {mark.label}
                </Text>
              </SliderMark>
            );
          })}
      </SliderRoot>
    );
  },
);

Slider.displayName = 'Slider';
export default React.memo(Slider);
