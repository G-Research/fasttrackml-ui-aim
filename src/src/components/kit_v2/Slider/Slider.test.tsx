import { render } from '@testing-library/react';

import Slider from './Slider';

global.ResizeObserver = require('resize-observer-polyfill');

// Test of Slider component
describe('<Slider />', () => {
  test('Renders Correctly', () => {
    const { asFragment } = render(<Slider />);
    expect(asFragment()).toMatchSnapshot();
  });

  // test of change the value props
  test('Change the value props', () => {
    const { asFragment, rerender } = render(<Slider />);
    expect(asFragment()).toMatchSnapshot();
    rerender(<Slider value={[10]} />);
    expect(asFragment()).toMatchSnapshot();
  });

  // test of change the defaultValue props
  test('Change the defaultValue props', () => {
    const { asFragment, rerender } = render(<Slider />);
    expect(asFragment()).toMatchSnapshot();
    rerender(<Slider defaultValue={[10]} />);
    expect(asFragment()).toMatchSnapshot();
  });

  // test of change the min and max props
  test('Change the min and max props', () => {
    const { asFragment, rerender } = render(<Slider />);
    expect(asFragment()).toMatchSnapshot();
    rerender(<Slider min={0} max={100} />);
    expect(asFragment()).toMatchSnapshot();
  });

  // test of render the marks props
  test('Render the marks props', () => {
    const { asFragment, rerender } = render(<Slider />);
    expect(asFragment()).toMatchSnapshot();
    rerender(
      <Slider
        marks={[
          {
            value: 0,
          },
          {
            value: 25,
          },
          {
            value: 50,
            label: '50',
          },
        ]}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
