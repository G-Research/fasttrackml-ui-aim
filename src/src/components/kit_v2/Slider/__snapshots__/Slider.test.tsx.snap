// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> Change the defaultValue props 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="c-jNrgAO"
    data-orientation="horizontal"
    data-testid="slider"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="c-jfCizU SliderTrack"
      data-orientation="horizontal"
    >
      <span
        class="c-xqWKh SliderRange"
        data-orientation="horizontal"
        style="left: 0%; right: 90%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(10% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="10"
        class="c-erhIZu SliderThumb"
        data-orientation="horizontal"
        data-radix-collection-item=""
        role="slider"
        style=""
        tabindex="0"
      >
        <span
          class="c-jbSxsI SliderLabel"
        >
          10
        </span>
      </span>
    </span>
  </span>
</DocumentFragment>
`;

exports[`<Slider /> Change the defaultValue props 2`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="c-jNrgAO"
    data-orientation="horizontal"
    data-testid="slider"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="c-jfCizU SliderTrack"
      data-orientation="horizontal"
    >
      <span
        class="c-xqWKh SliderRange"
        data-orientation="horizontal"
        style="left: 0%; right: 90%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(10% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="10"
        class="c-erhIZu SliderThumb"
        data-orientation="horizontal"
        data-radix-collection-item=""
        role="slider"
        style=""
        tabindex="0"
      >
        <span
          class="c-jbSxsI SliderLabel"
        >
          10
        </span>
      </span>
    </span>
  </span>
</DocumentFragment>
`;

exports[`<Slider /> Change the min and max props 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="c-jNrgAO"
    data-orientation="horizontal"
    data-testid="slider"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="c-jfCizU SliderTrack"
      data-orientation="horizontal"
    >
      <span
        class="c-xqWKh SliderRange"
        data-orientation="horizontal"
        style="left: 0%; right: 90%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(10% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="10"
        class="c-erhIZu SliderThumb"
        data-orientation="horizontal"
        data-radix-collection-item=""
        role="slider"
        style=""
        tabindex="0"
      >
        <span
          class="c-jbSxsI SliderLabel"
        >
          10
        </span>
      </span>
    </span>
  </span>
</DocumentFragment>
`;

exports[`<Slider /> Change the min and max props 2`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="c-jNrgAO"
    data-orientation="horizontal"
    data-testid="slider"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="c-jfCizU SliderTrack"
      data-orientation="horizontal"
    >
      <span
        class="c-xqWKh SliderRange"
        data-orientation="horizontal"
        style="left: 0%; right: 90%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(10% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="10"
        class="c-erhIZu SliderThumb"
        data-orientation="horizontal"
        data-radix-collection-item=""
        role="slider"
        style=""
        tabindex="0"
      >
        <span
          class="c-jbSxsI SliderLabel"
        >
          10
        </span>
      </span>
    </span>
  </span>
</DocumentFragment>
`;

exports[`<Slider /> Change the onValueChange props 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="c-jNrgAO"
    data-orientation="horizontal"
    data-testid="slider"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="c-jfCizU SliderTrack"
      data-orientation="horizontal"
    >
      <span
        class="c-xqWKh SliderRange"
        data-orientation="horizontal"
        style="left: 0%; right: 90%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(10% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="10"
        class="c-erhIZu SliderThumb"
        data-orientation="horizontal"
        data-radix-collection-item=""
        role="slider"
        style=""
        tabindex="0"
      >
        <span
          class="c-jbSxsI SliderLabel"
        >
          10
        </span>
      </span>
    </span>
  </span>
</DocumentFragment>
`;

exports[`<Slider /> Change the onValueChange props 2`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="c-jNrgAO"
    data-orientation="horizontal"
    data-testid="slider"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="c-jfCizU SliderTrack"
      data-orientation="horizontal"
    >
      <span
        class="c-xqWKh SliderRange"
        data-orientation="horizontal"
        style="left: 0%; right: 90%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(10% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="10"
        class="c-erhIZu SliderThumb"
        data-orientation="horizontal"
        data-radix-collection-item=""
        role="slider"
        style=""
        tabindex="0"
      >
        <span
          class="c-jbSxsI SliderLabel"
        >
          10
        </span>
      </span>
    </span>
  </span>
</DocumentFragment>
`;

exports[`<Slider /> Change the value props 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="c-jNrgAO"
    data-orientation="horizontal"
    data-testid="slider"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="c-jfCizU SliderTrack"
      data-orientation="horizontal"
    >
      <span
        class="c-xqWKh SliderRange"
        data-orientation="horizontal"
        style="left: 0%; right: 90%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(10% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="10"
        class="c-erhIZu SliderThumb"
        data-orientation="horizontal"
        data-radix-collection-item=""
        role="slider"
        style=""
        tabindex="0"
      >
        <span
          class="c-jbSxsI SliderLabel"
        >
          10
        </span>
      </span>
    </span>
  </span>
</DocumentFragment>
`;

exports[`<Slider /> Change the value props 2`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="c-jNrgAO"
    data-orientation="horizontal"
    data-testid="slider"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="c-jfCizU SliderTrack"
      data-orientation="horizontal"
    >
      <span
        class="c-xqWKh SliderRange"
        data-orientation="horizontal"
        style="left: 0%; right: 90%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(10% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="10"
        class="c-erhIZu SliderThumb"
        data-orientation="horizontal"
        data-radix-collection-item=""
        role="slider"
        style=""
        tabindex="0"
      >
        <span
          class="c-jbSxsI SliderLabel"
        >
          10
        </span>
      </span>
    </span>
  </span>
</DocumentFragment>
`;

exports[`<Slider /> Render the marks props 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="c-jNrgAO"
    data-orientation="horizontal"
    data-testid="slider"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="c-jfCizU SliderTrack"
      data-orientation="horizontal"
    >
      <span
        class="c-xqWKh SliderRange"
        data-orientation="horizontal"
        style="left: 0%; right: 90%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(10% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="10"
        class="c-erhIZu SliderThumb"
        data-orientation="horizontal"
        data-radix-collection-item=""
        role="slider"
        style=""
        tabindex="0"
      >
        <span
          class="c-jbSxsI SliderLabel"
        >
          10
        </span>
      </span>
    </span>
  </span>
</DocumentFragment>
`;

exports[`<Slider /> Render the marks props 2`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="c-jNrgAO"
    data-orientation="horizontal"
    data-testid="slider"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="c-jfCizU SliderTrack"
      data-orientation="horizontal"
    >
      <span
        class="c-xqWKh SliderRange"
        data-orientation="horizontal"
        style="left: 0%; right: 90%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(10% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="10"
        class="c-erhIZu SliderThumb"
        data-orientation="horizontal"
        data-radix-collection-item=""
        role="slider"
        style=""
        tabindex="0"
      >
        <span
          class="c-jbSxsI SliderLabel"
        >
          10
        </span>
      </span>
    </span>
    <span
      class="c-lmDkHl c-lmDkHl-iwHen-css"
      data-active="true"
    >
      <span
        class="c-iuedIb c-iuedIb-ijPcYZy-css"
      />
    </span>
    <span
      class="c-lmDkHl c-lmDkHl-ibdNZrc-css"
      data-active="false"
    >
      <span
        class="c-iuedIb c-iuedIb-ijPcYZy-css"
      />
    </span>
    <span
      class="c-lmDkHl c-lmDkHl-igXDGmT-css"
      data-active="false"
    >
      <span
        class="c-iuedIb c-iuedIb-ijPcYZy-css"
      >
        50
      </span>
    </span>
  </span>
</DocumentFragment>
`;

exports[`<Slider /> Renders Correctly 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="c-jNrgAO"
    data-orientation="horizontal"
    data-testid="slider"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="c-jfCizU SliderTrack"
      data-orientation="horizontal"
    >
      <span
        class="c-xqWKh SliderRange"
        data-orientation="horizontal"
        style="left: 0%; right: 90%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(10% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="10"
        class="c-erhIZu SliderThumb"
        data-orientation="horizontal"
        data-radix-collection-item=""
        role="slider"
        style=""
        tabindex="0"
      >
        <span
          class="c-jbSxsI SliderLabel"
        >
          10
        </span>
      </span>
    </span>
  </span>
</DocumentFragment>
`;
